### PR TITLE
feat: avoid node built-ins (path and url)

### PR DESCRIPTION
**Pull request by stefan-guggisberg**
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Thanks for contributing!

### CODE REVIEW

File: src/utils/path.js [hunk 1]
```diff
@@ -101,3 +101,14 @@ export function validatePathInfo(info) {
...
+
+/**
+ * Returns the file extension (e.g. '.html') of the given parh
+ * @param {string} path the path to get the extension from
+ * @return {string} the file extension or '' if there's no extension
+ */
+export function getExtension(path) {
+  const basename = path.split('/').pop();
+  const pos = basename.lastIndexOf('.');
+  return (basename === '' || pos < 1) ? '' : basename.slice(pos);
```
Comment (tripodsan) [inline src/utils/path.js:113]: this looks very expensive :-) also, it is only used in the foldermapping to check if there is an extension. so I would just move it there....
```suggestion
  const lastSlash = basename.lastIndexOf('/');
  const lastDot = basename.lastIndexOf('.', lastSlash);
  return lastDot >= 0 ? basename.substring(lastDot) : '';
```

File: test/steps/utils.test.js [hunk 2]
```diff
@@ -70,6 +70,11 @@ describe('Testing wrapNodes', () => {
 
 describe('Optimize Image URLs', () => {
   it('creates correct image optimize urls', () => {
+    assert.throws(() => optimizeImageURL(313), {
+      name: 'TypeError',
+      message: /^Parameter 'url' must be a string/,
+    });
```
Comment (tripodsan) [inline test/steps/utils.test.js:76]: doesn't t work like this?

```suggestion
    assert.throws(() => optimizeImageURL(313), new TypeError("Parameter 'url' must be a string"));
```

**Review by tripodsan — APPROVED** [review]
ok, but could be improved, imo

File: test/steps/utils.test.js (hunk 2 above)
Comment (stefan-guggisberg) [reply test/steps/utils.test.js:76]: No, the error message dynamic and matches a substring.

**Pull request merged**